### PR TITLE
Remove usageIds attribute, and use goId instead.

### PR DESCRIPTION
**Pull request by rantunesboreas**
Remove usageIds attribute, and use goId instead.

See issue: https://www.ebi.ac.uk/panda/jira/browse/GOA-2358

### CODE REVIEW

File: annotation-rest/src/main/java/uk/ac/ebi/quickgo/annotation/model/AnnotationRequest.java [hunk 1]
```diff
@@ -48,7 +47,6 @@
             GENE_PRODUCT_ID,
```
Comment (eddturner) [inline annotation-rest/src/main/java/uk/ac/ebi/quickgo/annotation/model/AnnotationRequest.java]: Just as a little housekeeping, do you think it'd be clearer if we renamed `TARGET_FIELDS` to something that showed intent better -- maybe like, `FILTER_REQUEST_FIELDS`?

File: annotation-rest/src/test/java/uk/ac/ebi/quickgo/annotation/model/AnnotationRequestTest.java [hunk 2]
```diff
@@ -132,9 +132,9 @@ public void setAndGetUsage() {
     public void setAndGetUsageIds() {
```
Comment (eddturner) [inline annotation-rest/src/test/java/uk/ac/ebi/quickgo/annotation/model/AnnotationRequestTest.java]: `setAndGetUsageIds` -> `setAndGetGOIds` ?

File: annotation-rest/src/test/java/uk/ac/ebi/quickgo/annotation/model/AnnotationRequestTest.java [hunk 3]
```diff
@@ -149,14 +149,14 @@ public void setAndGetUsageRelationships() {
     @Test
     public void createsFilterWithCaseInsensitiveUsageAndUsageIds() {
```
Comment (eddturner) [inline annotation-rest/src/test/java/uk/ac/ebi/quickgo/annotation/model/AnnotationRequestTest.java]: `createsFilterWithCaseInsensitiveUsageAndUsageIds` -> `createsFilterWithCaseInsensitiveUsageAndGOIds` ?

File: annotation-rest/src/test/java/uk/ac/ebi/quickgo/annotation/model/AnnotationRequestTest.java [hunk 4]
```diff
@@ -166,17 +166,17 @@ public void createsFilterWithCaseInsensitiveUsageAndUsageIds() {
     @Test
     public void createsFilterWithCaseInsensitiveUsageAndUsageIdsAndUsageRelationships() {
```
Comment (eddturner) [inline annotation-rest/src/test/java/uk/ac/ebi/quickgo/annotation/model/AnnotationRequestTest.java]: `createsFilterWithCaseInsensitiveUsageAndUsageIdsAndUsageRelationships` -> `createsFilterWithCaseInsensitiveUsageAndGOIdsAndUsageRelationships` ?

File: annotation-rest/src/main/java/uk/ac/ebi/quickgo/annotation/model/AnnotationRequest.java (hunk 1 above)
Comment (rantunesboreas) [inline annotation-rest/src/main/java/uk/ac/ebi/quickgo/annotation/model/AnnotationRequest.java]: Yes I do

File: annotation-rest/src/test/java/uk/ac/ebi/quickgo/annotation/model/AnnotationRequestTest.java (hunk 2 above)
Comment (rantunesboreas) [inline annotation-rest/src/test/java/uk/ac/ebi/quickgo/annotation/model/AnnotationRequestTest.java]: Done

File: annotation-rest/src/test/java/uk/ac/ebi/quickgo/annotation/model/AnnotationRequestTest.java (hunk 4 above)
Comment (rantunesboreas) [inline annotation-rest/src/test/java/uk/ac/ebi/quickgo/annotation/model/AnnotationRequestTest.java]: Done

File: annotation-rest/src/test/java/uk/ac/ebi/quickgo/annotation/model/AnnotationRequestTest.java (hunk 3 above)
Comment (rantunesboreas) [inline annotation-rest/src/test/java/uk/ac/ebi/quickgo/annotation/model/AnnotationRequestTest.java]: Done

**Pull request merged**